### ISSUE TITLE
Comparison of calculations with different units of measurement

### ISSUE THREAD

**Issue by Mikhail-Vaganov** [JuliaDiffEq/NBodySimulator.jl#4]
@dextorious , have a look at this figures for SI and OpenMM units usage. I tested simulation of 216 particles of liquid argon for 15 ps with timestep at  0.5 fs without thermostating. 

![rdf openmm and si](https://user-images.githubusercontent.com/16945627/43294028-caa95e88-9156-11e8-9ecb-cd4b9a9b1dde.png)
![temperature openmm and si](https://user-images.githubusercontent.com/16945627/43294033-d063ab58-9156-11e8-8ea6-4dea10f95e56.png)
![msd openmm and si](https://user-images.githubusercontent.com/16945627/43294037-d52da1d4-9156-11e8-804c-be17776bacef.png)

I didn't try to set the same initial configurations for both tests, though it is worth to do so for the sake of clearnes of the experiment/test.

Can you suggest other test systems with already implemented features for comparison calculations in different units? Berendsen thermostat can be used, for example, because it is deterministic.  
Well, at least, I can repeat such calculations for water systems.

**Comment by dextorious** [comment]
Right, the comparison has to be deterministic (the alternative would be running a statistically representative number of simulations, which would be prohibitively expensive at the moment). That pretty much just leaves liquid argon / water with identical initial conditions and either no thermostat or the Berendsen thermostat. I would also suggest noting down the times of each computation just in case  - NaNs/Infs can lead to slower performance, though hopefully we'll never see it happen.

**Comment by Mikhail-Vaganov** [comment]
And here are results for the water SPC/Fw simulations.

I used initial coordinates from the file **dextorious** provided me once (output_4.pdb). 3 ps at 0.5 fs step. 216 molecules without thermostating. 
![rdf for water openmm and si without thermostat](https://user-images.githubusercontent.com/16945627/43356840-918f6f48-9291-11e8-8964-bfb60f41acc4.png)
![temperature for water openmm and si without thermostat](https://user-images.githubusercontent.com/16945627/43356843-996116fe-9291-11e8-8dbf-2fcb1d26a4ec.png)
![msd for water openmm and si without thermostat](https://user-images.githubusercontent.com/16945627/43356848-ab17a480-9291-11e8-9024-07f4f3774bde.png)

**Comment by ChrisRackauckas** [comment]
Catastrophic cancellation is probably not happening because of a lot of the coefficient multiplications. Significance loss is really when you add/subtract values on different scales. OTOH, Multiplication/division is a safe operation. So the units must be fine enough because everything gets roughly scaled before the linear combinations of the integrator. 

But @dextorious does bring up a good point that the scaling would have a bigger effect on Float32 and GPUs, where multiplication/division by `6.022e23` could definitely hit the Float32 limits:

```julia
julia> 2.5f38
2.5f38

julia> 2.5f39
ERROR: syntax: overflow in numeric constant "2.5f39"
```

**Comment by dextorious** [comment]
Yeah, it looks like double precision is enough at least in this particular case, so this is about future proofing for Float32s as well as abiding by the conventions of the field for simplicity. I do think the future proofing part is fairly important in this case, since practical MD work is virtually always done on accelerators nowadays, most of which benefit hugely from reducing floating point precision.
